### ISSUE TITLE
`OSSL_STORE_load()` loading certs can be mislead to prompt for password

### ISSUE THREAD

**Issue by DDvO** [openssl/openssl#16224]
In case an (e.g., PEM formatted) certificate input file includes an encrypted key,
OSSL_STORE emits an unexpected and useless password prompt for it.

Here is a simple example, where the user needs to hit the Enter key for the `openssl verify` command to complete.
```
cat test/certs/root-cert.pem test/certs/key-pass-12345.pem >tmp.pem
apps/openssl verify -trusted tmp.pem test/certs/root-cert.pem
Enter pass phrase for tmp.pem:
test/certs/root-cert.pem: OK
```

Here is the stack trace for it:
```
Thread #1 [openssl] 20296 [core: 6] (Suspended : Signal : SIGINT:Interrupt)	
	_IO_fgets() at iofgets.c:53 0x7ffff7886feb	
	read_string_inner() at ui_openssl.c:339 0x7ffff7d17481	
	read_string() at ui_openssl.c:235 0x7ffff7d1729c	
	ui_read() at apps_ui.c:56 0x48deef	
	UI_process() at ui_lib.c:516 0x7ffff7d165f4	
	do_ui_passphrase() at passphrase.c:173 0x7ffff7c7b0a5	
	ossl_pw_get_passphrase() at passphrase.c:270 0x7ffff7c7b4a1	
	ossl_pw_passphrase_callback_dec() at passphrase.c:325 0x7ffff7c7b728	
	ossl_pw_get_passphrase() at passphrase.c:231 0x7ffff7c7b2e6	
	ossl_pw_passphrase_callback_dec() at passphrase.c:325 0x7ffff7c7b728	
	epki2pki_decode() at decode_epki2pki.c:91 0x7ffff7d7d833	
	decoder_process() at decoder_lib.c:936 0x7ffff7c05af8	
	pem2der_decode() at decode_pem2der.c:199 0x7ffff7d7e894	
	decoder_process() at decoder_lib.c:936 0x7ffff7c05af8	
	OSSL_DECODER_from_bio() at decoder_lib.c:80 0x7ffff7c0416f	
	file_load_file() at file_store.c:513 0x7ffff7db2be9	
	file_load() at file_store.c:680 0x7ffff7db3203	
	OSSL_STORE_load() at store_lib.c:428 0x7ffff7d085c2	
	load_key_certs_crls_suppress() at apps.c:966 0x488d46	
	load_key_certs_crls() at apps.c:1,093 0x4892fa	
	load_certs() at apps.c:798 0x48866f	
	verify_main() at verify.c:157 0x4818d0	
	do_cmd() at openssl.c:414 0x44ddc0	
	main() at openssl.c:295 0x44d9ef	
```

**Comment by t8m** [comment]
See #16228 for the fix/workaround.

**Comment by t8m** [comment]
Apparently with the current codebase this will be quite hard to fix.

**Comment by mattcaswell** [comment]
OTC: This should either be documented or fixed for 3.0 (if the scope of the fix is not fundamental)

**Comment by paulidale** [comment]
OTC: this is not a blocker for 3.0.  However, a fix for this will be included if available.
